### PR TITLE
NewConfig routine code copied.

### PR DESCRIPTION
**Pull request by SkyWolf369**
Should solve #266

### CODE REVIEW

File: gidari.go [hunk 1]
```diff
@@ -46,17 +47,37 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
 }
 
 // TransportFile will construct the transport operation using a configuration YAML file.
-func TransportFile(ctx context.Context, file *os.File) error {
-	cfg, err := NewConfig(ctx, file)
+func TransportFile(ctx context.Context, file *os.File) (*Config, error) {
```
Comment (prestonvasquez) [inline gidari.go:50]: We don't want this to return `*Config`, just the error.

File: gidari.go [hunk 2]
```diff
@@ -46,17 +47,37 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
+	// Disable logger
+	cfg.Logger.SetOutput(io.Discard)
+
+	return &Config{*cfg}, nil
+	if err != nil {
+		return nil, fmt.Errorf("unable to create new config: %w", err)
 	}
 
-	return Transport(ctx, cfg)
+	return nil, Transport(ctx, cfg)
```
Comment (prestonvasquez) [inline gidari.go:76]:
```suggestion
	return Transport(ctx, cfg)
```

File: gidari.go [hunk 3]
```diff
@@ -46,17 +47,37 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
 }
 
 // Transport will construct the transport operation using a "transport.Config" object.
-func Transport(ctx context.Context, cfg *Config) error {
+func Transport(ctx context.Context, cfg *transport.Config) error {
```
Comment (prestonvasquez) [inline gidari.go:80]: We can't change, transport is an internal package.
```suggestion
func Transport(ctx context.Context, cfg *Config) error {
```

**Review by prestonvasquez — CHANGES_REQUESTED** [review]
Looking good, few changes needed.

**Comment by SkyWolf369** [comment]
There you go.

**Comment by SkyWolf369** [comment]
Let me know if we are good.

File: gidari.go [hunk 4]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
 }
 
 // TransportFile will construct the transport operation using a configuration YAML file.
-func TransportFile(ctx context.Context, file *os.File) error {
-	cfg, err := NewConfig(ctx, file)
+func TransportFile(ctx context.Context, file *os.File) (error) {
+	info, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get file stat for reading: %w", err)
```
Comment (prestonvasquez) [inline gidari.go:53]:
```suggestion
		return fmt.Errorf("unable to get file stat for reading: %w", err)
```

File: gidari.go [hunk 5]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
 }
 
 // TransportFile will construct the transport operation using a configuration YAML file.
-func TransportFile(ctx context.Context, file *os.File) error {
-	cfg, err := NewConfig(ctx, file)
+func TransportFile(ctx context.Context, file *os.File) (error) {
```
Comment (prestonvasquez) [inline gidari.go:50]:
```suggestion
func TransportFile(ctx context.Context, file *os.File) error {
```

File: gidari.go [hunk 6]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
+	info, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get file stat for reading: %w", err)
+	}
+
+	bytes := make([]byte, info.Size())
+
+	_, err = file.Read(bytes)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file: %w", err)
```
Comment (prestonvasquez) [inline gidari.go:60]:
```suggestion
		return fmt.Errorf("unable to read file: %w", err)
```

File: gidari.go [hunk 7]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
+	}
+
+	cfg, err := transport.NewConfig(bytes)
 	if err != nil {
-		return fmt.Errorf("unable to create new config: %w", err)
+		return nil, fmt.Errorf("unable to create new config: %w", err)
```
Comment (prestonvasquez) [inline gidari.go:65]:
```suggestion
		return fmt.Errorf("unable to create new config: %w", err)
```

File: gidari.go [hunk 8]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
+	}
+
+	// Disable logger
+	cfg.Logger.SetOutput(io.Discard)
+
+	return &Config{*cfg}, nil
```
Comment (prestonvasquez) [inline gidari.go:71]:
```suggestion
```

File: gidari.go [hunk 9]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
+	if err != nil {
+		return nil, fmt.Errorf("unable to create new config: %w", err)
```
Comment (prestonvasquez) [inline gidari.go:73]:
```suggestion
		return fmt.Errorf("unable to create new config: %w", err)
```

File: gidari.go [hunk 10]
```diff
@@ -46,10 +47,30 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
...
 	}
 
 	return Transport(ctx, cfg)
```
Comment (prestonvasquez) [inline gidari.go:76]:
```suggestion
	return Transport(ctx, &Config{cfg})
```

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

**Comment by SkyWolf369** [comment]
Look good?

File: gidari.go [hunk 11]
```diff
@@ -21,6 +21,7 @@ type Config struct {
 	transport.Config
 }
 
+// TODO #265: Remove this routine
```
Comment (prestonvasquez) [inline gidari.go:24]:
```suggestion
// TODO #265: Remove this routine.
```

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

File: gidari.go [hunk 12]
```diff
@@ -21,6 +21,7 @@ type Config struct {
 	transport.Config
 }
 
+// TODO #265: Remove this routine.
```
Comment (prestonvasquez) [inline gidari.go:24]:
```suggestion
//noling:godox
// TODO #265: Remove this routine.
```

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

File: gidari.go [hunk 13]
```diff
@@ -21,6 +21,8 @@ type Config struct {
 	transport.Config
 }
 
+//noling:godox
```
Comment (prestonvasquez) [inline gidari.go:24]:
```suggestion
//nolint:godox
```

**Pull request merged**